backend/controller/observability/pubsub.go

```diff
@@ -0,0 +1,72 @@
+package observability
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/metric/noop"
+
+	"github.com/TBD54566975/ftl/backend/schema"
+	"github.com/TBD54566975/ftl/internal/observability"
+)
+
+const (
+	pubsubMeterName                = "ftl.pubsub"
+	pubsubTopicNameAttribute       = "ftl.pubsub.topic.name"
+	pubsubSubscriptionRefAttribute = "ftl.pubsub.subscription.ref"
+	pubsubSubscriberRefAttribute   = "ftl.pubsub.subscriber.sink.ref"
+)
+
+type PubSubMetrics struct {
+	meter            metric.Meter
+	published        metric.Int64Counter
+	subscriberCalled metric.Int64Counter
+}
+
+func initPubSubMetrics() (*PubSubMetrics, error) {
+	result := &PubSubMetrics{}
+	var errs error
+	var err error
+
+	result.meter = otel.Meter(pubsubMeterName)
+
+	counterName := fmt.Sprintf("%s.published", pubsubMeterName)
+	if result.published, err = result.meter.Int64Counter(
+		counterName,
+		metric.WithDescription("the number of times that an event is published to a topic")); err != nil {
+		result.published, errs = handleInitCounterError(errs, err, counterName)
+	}
+
+	counterName = fmt.Sprintf("%s.subscriber.called", pubsubMeterName)
+	if result.subscriberCalled, err = result.meter.Int64Counter(
+		counterName,
+		metric.WithDescription("the number of times that a pubsub event has been enqueued to asynchronously send to a subscriber")); err != nil {
+		result.subscriberCalled, errs = handleInitCounterError(errs, err, counterName)
+	}
+
+	return result, nil
```

Comment:
per the lint warning here, should we be returning the errors we get here? It looks like we always return `nil`

Reply:
fixed :)